src/python_testing/TC_IDM_1_4.py

```diff
@@ -0,0 +1,167 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from dataclasses import dataclass
+
+import chip.clusters as Clusters
+from chip import ChipUtility
+from chip.exceptions import ChipStackError
+from chip.interaction_model import InteractionModelError, Status
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main, type_matches
+from mobly import asserts
+
+
+@dataclass
+class FakeInvalidBasicInformationCommand(Clusters.BasicInformation.Commands.MfgSpecificPing):
+    @ChipUtility.classproperty
+    def must_use_timed_invoke(cls) -> bool:
+        return False
+
+
+class TC_IDM_1_4(MatterBaseTest):
+
+    @async_test_body
+    async def test_TC_IDM_1_4(self):
+        dev_ctrl = self.default_controller
+        dut_node_id = self.dut_node_id
+
+        self.print_step(0, "Commissioning - already done")
+
+        self.print_step(1, "Get remote node's MaxPathsPerInvoke")
+        session_parameters = dev_ctrl.GetRemoteSessionParameters(dut_node_id)
+        max_paths_per_invoke = session_parameters.maxPathsPerInvoke
+
+        asserts.assert_greater_equal(max_paths_per_invoke, 1, "Unexpected error returned from unsupported endpoint")
+        asserts.assert_less_equal(max_paths_per_invoke, 65535, "Unexpected error returned from unsupported endpoint")
+
+        self.print_step(2, "Sending `MaxPathsPerInvoke + 1` InvokeRequest if it fits into single MTU")
+        # In practice, it was noticed that we could only fit 57 commands before we hit the MTU limit as a result we
+        # conservatively try putting up to 100 commands into an Invoke request.
```

Comment:
so you're potentitally adding up to 100 commands, but only 57 fit into an MTU. But the description says this step only happens if it fits into an MTU. I'm not totally sure I get the logic here. Is there actually a check that this fits?

Reply:
Good catch. I will fix this. But to provide more context if you are curious

One of 2 things happens here:
1. Something changes that allows 100 commands to fit into the MTU. If this happens and DUT behaves properly we should assert fail in the unexpected success. But if the DUT unexpectedly fails we need to assert that number_of_commands_to_send != (max_paths_per_invoke + 1)
2. TH only fit <100 command before filling up MTU. This triggers the `except ChipStackError as e:` case where DUT claims to support more than we can fit. This results in the test step being skipped